packages/relay/src/lib/eth.ts

```diff
@@ -1146,28 +1146,59 @@ export class EthImpl implements Eth {
         throw e;
       }
     } else if (fromBlock || toBlock) {
-      const filters = [];
-      let order;
+      let fromBlockTimestamp;
+      let toBlockTimestamp;
+      let fromBlockNum = 0;
+      let toBlockNum;
+
       if (toBlock) {
-        // @ts-ignore
-        filters.push(`lte:${parseInt(toBlock)}`);
-        order = constants.ORDER.DESC;
+        try {
+          const blockResponse = await this.getHistoricalBlockResponse(toBlock, true, requestId);
+          toBlockTimestamp = blockResponse.timestamp.to;
+          toBlockNum = parseInt(blockResponse.number);
+        } catch (e) {
+          // Check if block error is RESOURCE_NOT_FOUND, the most likely scenario this will happen if a block number > height is passed
```

Comment:
Currently if toBlock number > height it will not set `toBlockTimestamp ` and `lte` timestamp filter, which means it will get logs up to the latest block. Then here https://github.com/hashgraph/hedera-json-rpc-relay/blob/9b845579ee4661eef4054b998f723f40c73f68da/packages/relay/src/lib/eth.ts#L1194-L1195 it fetches the latest block number just to check that the block range is inside of the set limit here https://github.com/hashgraph/hedera-json-rpc-relay/blob/9b845579ee4661eef4054b998f723f40c73f68da/packages/relay/src/lib/eth.ts#L1198-L1203

Reply:
Great that works.